README.md

```diff
@@ -1 +1,293 @@
+[![Dependency Status](https://david-dm.org/ipfs/js-ipfs-repo-migrations.svg?style=flat-square)](https://david-dm.org/ipfs/js-ipfs-repo-migrations)
+[![](https://img.shields.io/badge/made%20by-Protocol%20Labs-blue.svg?style=flat-square)](http://ipn.io)
+[![](https://img.shields.io/badge/project-IPFS-blue.svg?style=flat-square)](http://ipfs.io/)
+[![](https://img.shields.io/badge/freenode-%23ipfs-blue.svg?style=flat-square)](http://webchat.freenode.net/?channels=%23ipfs)
+[![standard-readme compliant](https://img.shields.io/badge/standard--readme-OK-green.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)
+[![js-standard-style](https://img.shields.io/badge/code%20style-standard-brightgreen.svg?style=flat-square)](https://github.com/feross/standard)
+![](https://img.shields.io/badge/npm-%3E%3D3.0.0-orange.svg?style=flat-square)
+![](https://img.shields.io/badge/Node.js-%3E%3D10.0.0-orange.svg?style=flat-square)
+
+> Migration framework for versioning of JS IPFS Repo
+
+This package is inspired by the [go-ipfs repo migration tool](https://github.com/ipfs/fs-repo-migrations/)
+
+## Lead Maintainer
+
+[Adam Uhlíř](https://github.com/auhau/)
+
+## Table of Contents
+
+- [Background](#background)
+- [Install](#install)
+  - [npm](#npm)
+  - [Use in Node.js](#use-in-nodejs)
+  - [Use in a browser with browserify, webpack or any other bundler](#use-in-a-browser-with-browserify-webpack-or-any-other-bundler)
+  - [Use in a browser Using a script tag](#use-in-a-browser-using-a-script-tag)
+- [Usage](#usage)
+  - [Writing migration](#writing-migration)
+  - [Migrations matrix](#migrations-matrix)
+- [API](#api)
+- [CLI](#cli)
+- [Versioning](#versioning)
+- [Contribute](#contribute)
+- [License](#license)
+
+## Background
+
+
+As js-ipfs evolves and new technologies, algorithms and data structures are incorporated it is necessary to 
+enable users to transition between versions. Different versions of js-ipfs may expect a different IPFS repo structure or content (see: [IPFS repo spec](https://github.com/ipfs/specs/tree/master/repo), [JS implementation](https://github.com/ipfs/js-ipfs-repo) ).
+So the IPFS repo is versioned, and this package provides a framework to create migrations to transition
+from one version of IPFS repo to the next/previous version.
+
+This framework:
+ * Handles locking/unlocking of repository
+ * Defines migrations API
+ * Executes and reports migrations in both directions: forward and backward
+ * Simplifies creation of new migrations
+
+## Install
+
+### npm
+
+```sh
+> npm install ipfs-repo-migrations
+```
+
+### Use in Node.js
+
+```js
+const migrations = require('ipfs-repo-migrations')
+```
+
+### Use in a browser with browserify, webpack or any other bundler
+
+```js
+const migrations = require('ipfs-repo-migrations')
+```
+
+## Usage
+
+Example:
+
+```js
+const migrations = require('ipfs-repo-migrations')
+const getVersion = require('ipfs-repo-migrations/repo/version')
```

Comment:
This is an actual antipattern. Teaching users to require specific files will make them fall on their face as a simple file name change can go unnoticed and break everyone's usage. Instead, use `migrations.getVersion`

Reply:
Fair point. Yet the `getVersion` function is there only for CLI sake and is not part of the main API. I wrote it here only to make the example more clear. I would suggest then only rewrite the example?

Reply:
For this example I'd just use `const currentRepoVersion = 7`